### ISSUE TITLE
Can I trigger forced sync of device calendar?

### ISSUE THREAD

**Issue by amrothemich** [lakshmanaprabhu/manage_calendar_events#10]
Hi!

I'm finding that the device calendar is really slow to update, even when Google Calendar on the device has received the update. I don't think this is a problem with the device_calendar package per se, but I wonder if you might know a way to trigger a refresh of the device calendar. For example, I created a new Google Calendar and it took 30 minutes before either package (device_calendar or manage_calendar_events) registered the change. Is there some way to trigger a refresh of the device calendar within my app?

https://stackoverflow.com/questions/67928544/device-calendar-in-flutter-slow-to-update

**Comment by lakshmanaprabhu** [comment]
Currently, The main purpose of the package is that to fetch the available calendars and events from the native device (iOS/Android) calendars. 
**If I understand correctly, you want to update your app calendar/event list whenever the native calendar gets the update.** 
To achieve that with manage_calendar_events, you should fetch the calendars/events **periodically** with the following APIs

- getCalendars() 
- getEvents()

**Comment by EmanueleVolpato** [comment]
@lakshmanaprabhu in ios work perfectly, but in android it takes too long to sync events

**Comment by amrothemich** [comment]
Hi,

Glad to hear it works well in iOS! I think the problem is that the device calendar itself isn't syncing often enough. 

I ended up using native code to force a sync of android device calendar. It's working okay, though deleted events and new calendars still take forever to show up.

In my flutter code:

`  static Future<void> syncCalendars() async {
    var status = await Permission.contacts.status;
    if (status.isDenied) {
      Permission.contacts.request();
    }
    const platform = MethodChannel('samples.flutter.dev/sync');
    try {
      await platform.invokeMethod('syncCalendars');
      print('successfully triggered sync');
    } on PlatformException catch (e) {
      print("Failed to sync calendars: '${e.message}'.");
    }
  }
`

and in main_activity:
`class MainActivity: FlutterActivity() {

    private val CHANNEL = "samples.flutter.dev/sync"

  override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
    super.configureFlutterEngine(flutterEngine)
    MethodChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL).setMethodCallHandler {
      // Note: this method is invoked on the main thread.
      call, result ->
      if (call.method == "syncCalendars") {
        print("here")
        result.success(syncCalendars())
    } else {
      print("Here")
        result.notImplemented()
      }
    }
  }
    private fun syncCalendars(): String{
        // AccountManager.newChooseAccountIntent()
        var accounts = AccountManager.get(context).getAccounts()
        // var authority = CalendarContract.Calendars.CONTENT_URI.getAuthority();
                // return accounts.toString()
        // Run query
        var authority: String = CalendarContract.Calendars.CONTENT_URI.getAuthority()!!
        for (i in 0..accounts.size - 1) {
          var extras: Bundle = Bundle()
          extras.putBoolean(ContentResolver.SYNC_EXTRAS_MANUAL, true);
          extras.putBoolean(ContentResolver.SYNC_EXTRAS_EXPEDITED, true);
          extras.putBoolean(ContentResolver.SYNC_EXTRAS_INITIALIZE, true);
          extras.putBoolean(ContentResolver.SYNC_EXTRAS_OVERRIDE_TOO_MANY_DELETIONS, true);
          extras.putBoolean(ContentResolver.SYNC_EXTRAS_REQUIRE_CHARGING, false);
          extras.putBoolean(ContentResolver.SYNC_EXTRAS_DO_NOT_RETRY, false);
          extras.putBoolean(ContentResolver.SYNC_EXTRAS_IGNORE_BACKOFF, true);
          // extras.putBoolean(ContentResolver.SYNC_EXTRAS_OVERRIDE_TOO_MANY_DELETIONS, true);
          Log.d("MyActivity", "Refreshing calendars for: " + accounts[i]);
          ContentResolver.requestSync(accounts[i], authority, extras);
        }
        print("here");
        
        return accounts[0].toString()

    }


}
`

**Comment by EmanueleVolpato** [comment]
yes you are right, after several tests it is not a problem with the library but with the calendar of the device, great job thanks
